packages/runtime/container-runtime/src/summary/summarizerTypes.ts

```diff
@@ -243,22 +245,21 @@ export type SummarizeResultPart<TSuccess, TFailure = undefined> =
 	  }
 	| {
 			success: false;
-			data: TFailure | undefined;
```

Comment:
What was the decision behind changing this?

Reply:
There was something failing in an intermediate state of my change. Looks like this isn't needed anymore. Reverted it back.

Reply:
Oh man! This is due to the annoying plumbing when summarization fails all the way from container runtime to running summarizer. The failures result transitions couple of times before eventually reaching there.

`ISummarizeResult` contains 3 promises - `summarySubmitted`, `summaryOpBroadcasted` and `receivedSummaryAckOrNack`. When summarization fails, the data for both `summarySubmitted` and `receivedSummaryAckOrNack` are set today in `SummarizeResultBuilder::fail`.
However, only one of them should be set on failure. With the current typing,